### DISCUSSION TITLE
Replace pings to wan for connection test with qbit api connection if possible

### DISCUSSION THREAD

**Discussion by aclerici38** [Feramance/qBitrr#137] [Ideas]
Hi there, would it be possible to replace the behavior of pinging the web for connection status with a get call to [/api/v2/transfer/info from the qbit api](https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)#transfer-info)? I think this is not only cleaner but handles cases where some torrents can be considered stale just because of a qbit restart, faulty vpn connection, etc. If it is not possible and I'm missing another use for the pings no problem. Thank you!

**Comment by Feramance** [comment]
The only point for the ping is to check whether torrents should be processed as normal or not. Therefore in the event no internet is detected, torrents are treated differently to account for the fact they might not be downloading. Ideally if qbittorrent is inside a VPN, qbitrr should be too to properly consider the different network. If the api call you suggested can do a similar check through qbittorrent directly, then I would consider that

**Reply by aclerici38** [reply]
Afaict it does. I linked the qbit documentation above

**Reply by Feramance** [reply]
Yes I'm taking a look in fact

**Reply by Feramance** [reply]
I've implemented the change in the latest version. It considers a valid connection if either one of the URLs, or qbittorrent has a connection

**Reply by aclerici38** [reply]
Thank you for the quick update! I have a couple thoughts though
- In the code it looks like it only checks for "Connected" status. I think this should be checking for "connected" or "firewalled". ~~Afaik the status will only return "connected" if there is a valid port forward and inbound connections to qbit, so if there are no active downloads I believe the status will be "firewalled" no matter what.~~ Edit: That doesn't seem to be the case. I'm not sure what the exact criteria for "firewalled" is, but I don't think it should be considered an invalid connection status.

- I don't see the point of having a valid connection if there's a connection to qbit OR the internet. I was hoping for a valid connection to be considered if both are true, I don't see a point in having qBitrr run if qbittorrent is not running/healthy.

**Reply by Feramance** [reply]
The reason I used an or is so that in case someone blocks internet access for qBitrr but allows it for qbittorrent, the appropriate checks will proceed. And with regards to the firewalled status, I will adjust it to include it.

**Reply by aclerici38** [reply]
ok, would you be open to adding a config flag or something instead then? Currently it's not possible to run qbitrr without an internet connection so, while it's nice to support, I think that's a pretty significant edge case.

**Reply by Feramance** [reply]
Sorry for the late reply, no config for now but things should be running smoothly as is
